pkg/services/install/install_test.go

```diff
@@ -0,0 +1,524 @@
+			Expect(err).ShouldNot(HaveOccurred())
+
+			systemKustomization := models.CreateKustomization(clusterName, testNamespace, models.RuntimePath, models.SourcePath, models.SystemKustResourcePath, models.UserKustResourcePath, models.WegoAppPath)
+
+			systemKustomizationManifest, err := yaml.Marshal(systemKustomization)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			source, err := models.GetSourceManifest(context.Background(), fakeFluxClient, fakeGitProvider, clusterName, testNamespace, configRepo, "main")
+			Expect(err).ShouldNot(HaveOccurred())
+
+			expectedManifests := []models.Manifest{
+				{
+					Path:    git.GetSystemQualifiedPath(clusterName, models.AppCRDPath),
+					Content: manifests.AppCRD,
+				},
+				{
+					Path:    git.GetSystemQualifiedPath(clusterName, models.RuntimePath),
+					Content: runtimeManifests,
+				},
+				{
+					Path:    git.GetSystemQualifiedPath(clusterName, models.SystemKustResourcePath),
+					Content: systemKustomizationResource,
+				},
+				{
+					Path:    git.GetSystemQualifiedPath(clusterName, models.UserKustResourcePath),
+					Content: userKustomizationResource,
+				},
+				{
+					Path:    git.GetSystemQualifiedPath(clusterName, models.WegoAppPath),
+					Content: wegoAppManifest,
+				},
+				{
+					Path:    git.GetSystemQualifiedPath(clusterName, models.WegoConfigPath),
+					Content: wegoConfigManifest,
+				},
+				source,
+				{
+					Path:    git.GetSystemQualifiedPath(clusterName, models.SystemKustomizationPath),
+					Content: systemKustomizationManifest,
+				},
+				{
+					Path:    filepath.Join(git.GetUserPath(clusterName), ".keep"),
+					Content: strconv.AppendQuote(nil, "# keep"),
+				},
+			}
+
+			applyIndex := 0
+			fakeKubeClient.ApplyCalls(func(ctx context.Context, manifest []byte, namespace string) error {
+
+				if applyIndex <= 6 {
+
+					Expect(namespace).Should(Equal(namespace))
+
+					if applyIndex != 0 {
+						partOfPreviousManifest := bytes.Contains(expectedManifests[applyIndex-1].Content, manifest)
+						if partOfPreviousManifest {
+							Expect(string(expectedManifests[applyIndex-1].Content)).Should(ContainSubstring(string(manifest)))
+							return nil
+						}
+					}
+
+					Expect(string(expectedManifests[applyIndex].Content)).Should(ContainSubstring(string(manifest)))
+
+					applyIndex++
+				}
+				return nil
+			})
+
+			writeIndex := 0
+			fakeGitClient.CloneReturns(true, nil)
+			fakeGitClient.WriteCalls(func(path string, content []byte) error {
+				if writeIndex < 8 {
+					Expect(path).Should(Equal(expectedManifests[writeIndex].Path))
+					Expect(string(content)).Should(Equal(string(expectedManifests[writeIndex].Content)))
+				}
+				writeIndex++
+				return nil
+			})
+
+			err = installer.Install(testNamespace, configRepo, true)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		It("should succeed with auto-merge=false", func() {
+			fakeKubeClient.GetClusterStatusReturns(kube.Unmodified)
```

Comment:
a quick glance and it appears that a lot of this is duplicated from the auto-merge=true test.  Could it be refactored into a common function and be more readable?